tc/sgx/trusted_worker_manager/enclave/kme/workload_processor_kme.h

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * Defines base class WorkloadProcessor and other definitions to
+ * create an Avalon workload processor.
+ * To use, #include "workload_processor.h"
+ */
+
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "ext_work_order_info_kme.h"
+#include "workload_processor.h"
+
+/** Class to register, create, and process a kme workload. */
+class WorkloadProcessorKME : public WorkloadProcessor {
+public:
+    WorkloadProcessorKME(void);
+    virtual ~WorkloadProcessorKME(void);
+
+    /**
+     * Register a WorkloadProcessor.
+     * Used by the workloads to register themselves
+     *
+     * @param workload_id Workload identifier
+     * @returns           Pointer to WorkloadProcessorKME
+     */
+    static WorkloadProcessorKME* RegisterWorkloadProcessorKME(
+        std::string workload_id, WorkloadProcessorKME* processor);
+
+    /** Mapping between workload id and KME WorkloadProcessor. */
+    static std::map<std::string, WorkloadProcessorKME*> workload_processor_table;
```

Comment:
Can we use base class table since `WorkloadProcessorKME` is derived from `WorkloadProcessor`

Reply:
Done